File: c#/str_join.cs
using System;
using System.Text;
using System.Diagnostics;

class Program {
    const string Chunk = "regexp123abcbcd";
    const int RepeatCount = 10000;
    const int Size = 1000;

    static void Join() {
        var sb = new StringBuilder();
        for (var i = 0; i < Size; ++i) sb.Append(Chunk);
        sb.ToString();
    }

    static void Main() {
        var sw = new Stopwatch();
        sw.Start();
        for (var i = 0; i < RepeatCount; ++i) Join();
        sw.Stop();
        Console.WriteLine(sw.ElapsedMilliseconds);
    }
}

File: nim/str_join.nim
import times
import strutils

const chunk = "regexp123abcbcd"
const repeatCount = 10000
const size = 1000

proc join(): void =
    var str = ""
    for i in 1..size:
        str.add(chunk)

let startTime = epochTime()
for i in 1..repeatCount:
    join()
let endTime = epochTime()

echo int((endTime - startTime) * 1000)
